internal/interpreter/api.rs

```diff
@@ -148,6 +168,151 @@ impl Value {
             _ => ValueType::Other,
         }
     }
+
+    /// Create a `Value` from a JSON Value
+    #[cfg(feature = "json")]
+    pub fn from_json(t: &i_slint_compiler::langtype::Type, v: &serde_json::Value) -> Option<Self> {
```

Comment:
Actually, this shouldn't take the langtype::Type, that type is internal and not exposed to the API.
We have slint_interpreter::ValueType, but it doens't contain all types and especially not the struct which we care about.

Reply:
that would break compatibility with the previous viewer though.
It is kind of neat that we don't need to encode the type in the json.

Reply:
Yes, that breaks compatibility with the old viewer but also gets rid of some magic. The type encoding does not feel too bad to me, having played with it for a bit.

Take a look, tell me what you think.

Reply:
I chose to encode things like this now:

`{ "image-path": "PATH" }`: Image

`{ "struct: { /* fields */ }`: Struct

`{ "color": "#abcdef01" }`: Color

`{ "enum": "Foo", "variant": "Bar" }`: Enum value